app/assets/javascripts/collections/story_collection.js

```diff
@@ -18,7 +18,49 @@ module.exports = Backbone.Collection.extend({
     this.labels = [];
   },
 
-  saveSorting: function(columnName) {
+  calculateNewPosition: function(previous_story_id, next_story_id) {
+    var newPosition;
+    if (!_.isUndefined(previous_story_id)) {
+      newPosition = this.calculatePositionAfter(previous_story_id);
+    } else if (!_.isUndefined(next_story_id)) {
+      newPosition = this.calculatePositionBefore(next_story_id);
+    } else {
+      if (this.length !== 1) {
+        throw "Unable to determine previous or next story id for dropped story";
+      }
+    }
+    return newPosition;
+  },
+
+  calculatePositionAfter: function(beforeId) {
+    var before = this.get(beforeId);
```

Comment:
What do you think about using `const` for the `before`, `after`, `difference`  and `newPosition` since you don't reassign them?